x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ilm/WaitForIndexColorStep.java

```diff
@@ -75,17 +75,26 @@ public boolean equals(Object obj) {
 
     @Override
     public Result isConditionMet(Index index, ClusterState clusterState) {
-        String indexName = indexNamePrefix != null ? indexNamePrefix + index.getName() : index.getName();
-        IndexMetadata indexMetadata = clusterState.metadata().index(index);
```

Comment:
The problem originated here as we don't use the computed `indexName` but the original `index`